File: algorithms/SelectionSort.cpp
#include "Driver.h"

void selectionSort(std::vector<int>& input)
{
	for (int i = 0; i < input.size() - 1; ++i) {
		int minIndex = i;
		for (int j = i; j < input.size(); ++i) {
			if (input[j] > input[minIndex]) {
				minIndex = j;
			}
		}
		std::swap(input[minIndex], input[i]);
	}
}

int main(int argn, char* argv[])
{
	runMethod(argn, argv, selectionSort);
	return 0;
}

File: algorithms/sorting/java/SelectionSort.java
class SelectionSort implements SortMethod
{
  @Override
  public void sort(int[] input)
  {
    for (int i = 0; i < input.length - 1; ++i) {
      int minIndex = i;
      for (int j = i; j < input.length; ++j) {
        if (input[j] < input[minIndex]) {
          minIndex = j;
        }
      }
      int temp = input[minIndex];
      input[minIndex] = input[i];
      input[i] = temp;
    }
  }

  public static void main(String[] argv)
  {
    Driver.run(new SelectionSort());
  }
}
